src/ast/passes/codegen_llvm.h

```diff
@@ -89,8 +89,17 @@ class CodegenLLVM : public Visitor
 
     ScopedExprDeleter(const ScopedExprDeleter &other) = delete;
     ScopedExprDeleter &operator=(const ScopedExprDeleter &other) = delete;
-    ScopedExprDeleter(ScopedExprDeleter &&other) = default;
-    ScopedExprDeleter &operator=(ScopedExprDeleter &&other) = default;
+
+    ScopedExprDeleter(ScopedExprDeleter &&other)
+    {
+      *this = std::move(other);
```

Comment:
is this just a shorthand that says "[move constructor](https://en.cppreference.com/w/cpp/language/move_constructor) delegates to [move assignment operator](https://en.cppreference.com/w/cpp/language/move_assignment)"?

Reply:
Yes. When move-constructing ScopedExprDeleter the compiler will first default initialize deleter_ to be an empty function and then the constructor will delegate to operator= to handle disarming of other.